### PR TITLE
Cross region creation of VM restore points

### PR DESCRIPTION
**Pull request by dikethir**
Cross region creation of VM restore points

### CODE REVIEW

File: Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+}
+```
+
+## Known issues and limitations
+* Cross-region copy functionality is only supported via REST APIs. ARM Template deployment, CLI, SDK, PS will be added later. 
+* Concurrent creation of Restore Points for a VM in the same or different regions is not supported
+* Error messages may not reflect the actual error that has occurred. Error messages will be updated.
+* Restore point of CMK encrypted VM and disks will be encrypted using PMK in the target region. Customer needs to use CMK when restoring the disks and VM from the restore point.
+* Currently, the replication progress in only updated once every 10mins. Hence for disks that have low churn, there can be scenarios where only the initial (0) and the terminal replication progress (100) can be seen.
+* If there is huge amount data to be copied to the target region, depending on the bandwidth available between the regions, the creation time could be couple of days. If the creation time exceeds 2 weeks, the restore points creation operation will be terminated.
```
Comment (shahja1) [inline Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md:453]: We are talking about "creation time"... Isn't it rather "copy time"?

File: Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+* When a disk restore point creation fails, its completion percent is shown as 0, not the intermediate value where it failed.
```
Comment (shahja1) [inline Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md:454]: Can you pls. confirm with Shrutika on this? I believe we made a change to not show "completion percent" at all in such failure scenarios.

File: Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md [hunk 3]
```diff
@@ -0,0 +1,457 @@
...
+* Restoring of Disk from Restore point does not automatically check if the disk restore points replication is completed. You need to manually check the replication status and start restoring the disk only after the disk restore points is replicated successfully.
```
Comment (shahja1) [inline Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md:455]: I feel this is also fixed now. Pls. confirm with Shrutika on the check being added in DiskRP for restore scenarios.

File: Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md [hunk 4]
```diff
@@ -0,0 +1,457 @@
...
+* Restore points that are copied to the target region do not have a reference to the source VM. The source VM reference is available in the Restore Point Collection
+* Other known issues and limitations of cross region copy of restore points may apply here as well
```
Comment (shahja1) [inline Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md:457]: We should add details about how the billing would be visible to the customers today in this section

**Review by shahja1 — CHANGES_REQUESTED** [review]

File: Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md (hunk 3 above)
Comment (dikethir) [reply Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md]: removed it.

File: Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md (hunk 2 above)
Comment (dikethir) [reply Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md]: removed it after confirmation

File: Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md (hunk 1 above)
Comment (dikethir) [reply Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md]: You are right. changed it.

File: Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md [hunk 5]
```diff
@@ -0,0 +1,457 @@
...
+            ]
+        }
+    }
+}
+```
+
+## Known issues and limitations
+* Cross-region copy functionality is only supported via REST APIs. ARM Template deployment, CLI, SDK, PS will be added later. 
+* Concurrent creation of Restore Points for a VM in the same or different regions is not supported
+* Error messages may not reflect the actual error that has occurred. Error messages will be updated.
```
[the remark this answers is not in the document]
Comment (dikethir) [reply Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md]: Agreed.

File: Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md [hunk 6]
```diff
@@ -0,0 +1,457 @@
...
+                    }
+                }
+            ]
+        }
+    }
+}
+```
+
+## Known issues and limitations
+* Cross-region copy functionality is only supported via REST APIs. ARM Template deployment, CLI, SDK, PS will be added later.
```
[the remark this answers is not in the document]
Comment (dikethir) [reply Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md]: Done

File: Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md [hunk 7]
```diff
@@ -0,0 +1,457 @@
+## Cross-region creation of VM Restore Points
+Ability to create VM Restore Points directly in the target region by referencing a VM in the source region
+**Scenario where this API can be helpful:** Implement a disaster recovery solution to protect VMs from region failure.
+
+### Create Restore Point Collection in target region
+First step in creating a VM Restore point in a target region referencing a VM from a source region is to create a RestorePointCollection in the target region.
+
+#### URI Request
+```
+PUT https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/restorePointCollections/{restorePointCollectionName}&api-version={api-version}
```
[the remark this answers is not in the document]
Comment (dikethir) [reply Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md]: Done. The design doc has this. We should probable update there as well.

File: Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md [hunk 8]
```diff
@@ -0,0 +1,457 @@
...
+    "type": "Microsoft.Compute/restorePointCollections",
+    "location": "location of the restorePointCollection resource",
+    "tags": {
+        "department": "finance"
+    },
+    "properties   ": {
+        "restorePointCollectionId": "Guid Id of RestorePointCollection",
+        "source": {
+            "id": "/subscriptions/{subid}/resourceGroups/{resourceGroupName}/providers/microsoft.compute/virtualMachines/{vmName}",
+            "location": "westus"
```
[the remark this answers is not in the document]
Comment (dikethir) [reply Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md]: Done

File: Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md [hunk 9]
```diff
@@ -0,0 +1,457 @@
+## Cross-region creation of VM Restore Points
```
[the remark this answers is not in the document]
Comment (dikethir) [reply Cross Region VM Restore Points/Cross Region Creation of VM Restore Points.md]: Agreed. Changed it.

File: Cross Region VM Restore Points/Cross Region Creation of Restore Points/Cross Region Creation of VM Restore Points.md [hunk 10]
```diff
@@ -0,0 +1,456 @@
...
+}
+```
+
+
+### Get VM Restore Points With Copy/Replication Status
+Once creation of VM Restore Points is initiated, you can track the data copy status by calling GET instance View (?$expand=instanceView) on the VM Restore Point.
+
+#### URI Request
+```
+GET https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/restorePointCollections/{restorePointCollectionName}/restorePoints/{restorePointName}?$expand=instanceView?api-version={api-version}
```
Comment (shahja1) [inline Cross Region VM Restore Points/Cross Region Creation of Restore Points/Cross Region Creation of VM Restore Points.md:308]: ?$expand=instanceView**?**api-version={api-version} is not correct... it should be "?$expand=instanceView**&**api-version={api-version}"

File: README.md [hunk 11]
```diff
@@ -11,15 +11,20 @@ A VM Restore Point stores the VM configuration and point-in-time crash (if the V
...
- 
+## Cross Region VM Restore Points
+As an extension to VM Restore Points we are providing additional functionality within Azure platform to enable our partners to build BCDR solutions for Azure VMs. Additional functionalities include: 
+**1. Ability to copy VM Restore Points from one region to another other region**
  Scenarios where this API can be helpful:
  * Extend multiple copies of backup to different regions
  * Extend local backup solutions to support disaster recovery from region failures
 
- For copying VM Restore Points across regions please refer to the [Cross Region VM Restore Points documentation](https://github.com/Azure/Virtual-Machine-Restore-Points/blob/main/Cross%20Region%20VM%20Restore%20Points/Cross%20Region%20Copy%20of%20VM%20Restore%20Points.md).
+ For copying VM Restore Points across regions please refer to the [Cross Region Copy of VM Restore Points documentation](https://github.com/Azure/Virtual-Machine-Restore-Points/blob/main/Cross%20Region%20VM%20Restore%20Points/Cross%20Region%20Copy%20of%20VM%20Restore%20Points.md).
```
Comment (shahja1) [inline README.md:21]: We need to update this link.

File: README.md [hunk 12]
```diff
@@ -11,15 +11,20 @@ A VM Restore Point stores the VM configuration and point-in-time crash (if the V
...
+
+**2. Ability to create VM Restore Points directly in the target region by referencing a VM in the source region**
+Scenario where this API can be helpful: 
+* Implement a disaster recovery solution to protect VMs from region failure.
+
+For creating VM Restore Points across regions please refer to the [Cross Region Creation of VM Restore Points documentation](https://github.com/Azure/Virtual-Machine-Restore-Points/blob/main/Cross%20Region%20VM%20Restore%20Points/Cross%20Region%20Creation%20of%20VM%20Restore%20Points.md).
```
Comment (shahja1) [inline README.md:27]: We need to update this link

File: README.md [hunk 13]
```diff
@@ -57,33 +54,29 @@ Use the following URI for GET and DELETE operation on the Restore Point resource
 https://management.azure.com/subscriptions/{SubscriptionID}/resourceGroups/{ResourceGroupName}/providers/Microsoft.Compute/restorePointCollections/{RestorePointCollectionName}/RestorePoints/{RestorePointName}?api-version=2021-03-01
  
 For update of an existing Restore Point Collection resource, only update of tags is permitted.
-## Create Disk
 
+## Create Disk
 1. Get the Disk Restore ID by either using GET call on Restore Point Collection and expanding Restore Points or by doing a call on Restore Point(s). 
-2. You can then use this ARM template to create a disk using diskRestorePoint: https://github.com/Azure/Virtual-Machine-Restore-Points/blob/main/createDiskFromDiskRestorePoint.json
```
Comment (shahja1) [inline README.md:63]: Why are we making changes to RestorePoints documentation?

File: README.md [hunk 14]
```diff
@@ -31,18 +36,10 @@ The VM Restore Point feature is currently in private preview and is not meant fo
...
 
 ## Creating a VM Restore Point
-1. First Step is to create a RestorePointCollection. You can do so by using the template https://github.com/Azure/Virtual-Machine-Restore-Points/blob/main/createRestorePointCollection.json
-2. Next, you can create a restore point using the RestorePointCollection. You can do so by using the template https://github.com/Azure/Virtual-Machine-Restore-Points/blob/main/CreateRestorePoint.json
-
-2a. You can also create a restore point and exclude one or more attached disks by using the following API call:
+1. First Step is to create a RestorePointCollection. You can refer to the following API documentation on [how to create a RestorePointCollection](https://docs.microsoft.com/en-us/rest/api/compute/restore-point-collections/create-or-update#create-or-update-a-restore-point-collection.).
+2. Next, you can create a restore point using the RestorePointCollection. You can refer to the following API documentation on [how to create a RestorePoint](https://docs.microsoft.com/en-us/rest/api/compute/restore-points/create#create-a-restore-point).
+* You can also create a restore point and exclude one or more attached disks.
```
Comment (shahja1) [inline README.md:42]: [same] Why are we making changes to RestorePoints documentation?

File: README.md [hunk 15]
```diff
@@ -57,33 +54,29 @@ Use the following URI for GET and DELETE operation on the Restore Point resource
...
-
 We are providing a BeginGetAccess API through which the user can directly pass the ID of the diskRestorePoint to create a SAS to the underlying disk. 
 
-   If there is no active SAS on the incremental snapshot of the restore point, then a new SAS will be created and Url returned to the user. 
-
-   If there is already an active SAS, the SAS will be extended and original SASUrl will be returned. 
+* If there is no active SAS on the incremental snapshot of the restore point, then a new SAS will be created and Url returned to the user. 
+* If there is already an active SAS, the SAS will be extended and original SASUrl will be returned. 
 
 POST
```
Comment (shahja1) [inline README.md:82]: Simply name the .docx files now "Api Documentation.docx" since they are already within the respective folders.

**Review by shahja1 — CHANGES_REQUESTED** [review]

File: Cross Region VM Restore Points/Cross Region Creation of Restore Points/Cross Region Creation of VM Restore Points.md (hunk 10 above)
Comment (dikethir) [reply Cross Region VM Restore Points/Cross Region Creation of Restore Points/Cross Region Creation of VM Restore Points.md]: Done

File: README.md (hunk 11 above)
Comment (dikethir) [reply README.md]: Done

File: README.md (hunk 12 above)
Comment (dikethir) [reply README.md]: Done

File: README.md (hunk 13 above)
Comment (dikethir) [reply README.md:63]: Restore Points team is not planning to update the GitHub repo anymore as they have the public documentation for APIs published for public preview. So added link to there public here for completeness.

File: README.md (hunk 14 above)
Comment (dikethir) [reply README.md:42]: Restore Points team is not planning to update the GitHub repo anymore as they have the public documentation for APIs published for public preview. So added link to there public here for completeness.

File: README.md (hunk 15 above)
Comment (dikethir) [reply README.md:82]: Done

**Review by shahja1 — APPROVED** [review]

**Pull request merged**